src/login/logind-session-dbus.c

```diff
@@ -938,7 +929,10 @@ int session_send_create_reply_bus(Session *s, const sd_bus_error *error) {
                         s->id,
                         p,
                         s->user->runtime_path,
-                        fifo_fd,
+                        s->leader.fd, /* This is supposed to be a fifo fd which older versions logind would
+                                         return to clients and subscribe to EOF upon which the session gets
+                                         closed. We've bumped our baseline to v5.4 where pidfd is better
+                                         suited for the job. But still need to return *something* to the client. */
```

Comment:
hmm, i'd prefer if we'd return something more "useless" here. who knows, maybe the client actually runs in a sandbox that otherwise disallows access to pidfd. now suddenly they can get one for themselves and do stuff with it, which might not be desired.

maybe just allocate a pipe here, and close one side already.

or maybe return an eventfd or so, as one of the most lightweight things we have i figure.

the good thing about eventfds is that (at least afaik) you cannot use them to gather info or evoke operations on any other object, everything you can do to them only applies to eventfds...